src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -153,17 +153,23 @@ private boolean decideAndUseCache(String remoteName) throws IOException, Interru
                 Git git = Git.with(TaskListener.NULL, new EnvVars(EnvVars.masterEnvVars)).in(cacheDir).using("jgit");
                 GitClient client = git.getClient();
                 if (client.hasGitRepo()) {
-                    sizeOfRepo = FileUtils.sizeOfDirectory(cacheDir);
-                    sizeOfRepo = (sizeOfRepo/1000); // Conversion from Bytes to Kilo Bytes
+                    long clientRepoSize = FileUtils.sizeOfDirectory(cacheDir) / 1024;
+                    if (clientRepoSize > sizeOfRepo) {
```

Comment:
In what event would we have multiple caches for the same git repository? 
Would it happen if two different multi branch projects work on a different version of the same git repository?

Reply:
I think that one case that might happen is two jobs that refer to a remote repository with different URLs and with different update frequencies.  If one of the caches is only updated once a week and the other is updated once an hour, the "once an hour" repository may have a larger size than the "once a week" repository.  This assumes the larger of the two values is the better approximation of repository size.